### DISCUSSION TITLE
site to site VPN with Palo Alto

### DISCUSSION THREAD

**Discussion by antai1** [strongswan/strongswan#534] [Q&A]
I have been trying to set up a S2S VPN using Palo Alto (on-site) and StrongSwan on AWS.

After configuring StrongSwan, I successfully connected from a client to the server.  I did that to test the config and to ensure traffic was connecting to AWS correctly.

Next, I configured the /etc/ipsec.secrets file for PSK and configured /etc/ipsec.conf with what I think are the correct parameters.

conn ikev2-vpn
    authby=secret
    auto=start
    type=tunnel
    keyexchange=ikev2
    compress=no
    left={internal IP Address of Server}
    leftid={external IP Address of Server}
    leftsubnet={Subnet of Server}
    leftfirewall=yes
    leftauth=psk
    right={external IP Address to PA}
    rightid=%any
    rightsubnet=0.0.0.0./0
    rightauth=psk
    ike=aes128-sha256-modp2048!
    esp=aes128-sha256-modp2048!
    ikelifetime=24h
    lifetime=24h
    mobike=no
    forceencaps=yes
    dpdaction=restart
    dpddelay=10

/etc/ipsec

#ikev2-vpn
{external ip address of onsite network}: PSK "Alpha Numeric number"

log results:

root@ip-{Internal IP Address of server}:~# ipsec up ikev2-vpn
initiating IKE_SA ikev2-vpn[38] to {IP Address of PA}
generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
sending packet: from {Internal IP Address of server}[500] to {IP Address of PA}[500] (464 bytes)
received packet: from {IP Address of PA}[500] to {Internal IP Address of server}[500] (432 bytes)
parsed IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) ]
selected proposal: IKE:AES_CBC_128/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
local host is behind NAT, sending keep alives
remote host is behind NAT
**sending cert request for "CN=VPN root CA"**
authentication of '{External IP Address of AWS Server}' (myself) with pre-shared key
establishing CHILD_SA ikev2-vpn{38}
generating IKE_AUTH request 1 [ IDi N(INIT_CONTACT) CERTREQ IDr AUTH SA TSi TSr N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
sending packet: from {Internal IP Address of server}[4500] to {IP Address of PA}[4500] (272 bytes)
received packet: from {IP Address of PA}[4500] to {Internal IP Address of server}[4500] (80 bytes)
parsed IKE_AUTH response 1 [ N(AUTH_FAILED) ]
received AUTHENTICATION_FAILED notify error
establishing connection 'ikev2-vpn' failed

My question is WHY is the connection attempt "**sending cert request for "CN=VPN root CA**""???  I am trying to use PSK.  Could this be the reason it is failing?

Thanks in advance.

**Comment by Thermi** [comment]
> conn ikev2-vpn
> authby=secret
> auto=start
> type=tunnel
> keyexchange=ikev2
> compress=no
> left={internal IP Address of Server}
> leftid={external IP Address of Server}
> leftsubnet={Subnet of Server}
> leftfirewall=yes
> leftauth=psk
> right={external IP Address to PA}
> rightid=%any
> rightsubnet=0.0.0.0./0
> rightauth=psk
> ike=aes128-sha256-modp2048!
> esp=aes128-sha256-modp2048!
> ikelifetime=24h
> lifetime=24h
> mobike=no
> forceencaps=yes
> dpdaction=restart
> dpddelay=10
> 
> My question is WHY is the connection attempt "**sending cert request for "CN=VPN root CA**""??? I am trying to use PSK. Could this be the reason it is failing?
> 
> Thanks in advance.

Unlikely. Ask the other peer's op for the reason.

Btw, you are lucky the connection failed because your configuration would break your SSH connection and also break it right away after the daemon start, potentially making the machine unreachable:

> leftsubnet={Subnet of Server}
> rightsubnet=0.0.0.0./0
> auto=start

You are currently configuring a policy based tunnel and you are requesting a TS that would force all packets from any local IP to any destination to go through the tunnel.

What exactly are you trying to configure?
[There are many example configs on the wiki for site-to-site connections.](https://wiki.strongswan.org/projects/strongswan/wiki/UsableExamples)

I propose you try to use an XFRM interface (preferred) or a VTI (better than policy based) for your site-to-site tunnel. That'd be way better for your use case.

**Comment by antai1** [comment]
This weekend I was just standing up a small replica of an instance another company is using that we are working with.  We have to get the tunnel between us up in order for me to push data through the tunnel for them to forward onto one of their servers.

I am behind a Palo Alto network and they are using StrongSwan for their VPN solution.  We have been tweaking the config between us for several days on and off and still end up with the same issue that 
"received AUTHENTICATION_FAILED notify error"
"establishing connection 'ikev2-VPN failed"

They are not seeing the issue reference "sending cert request for "CN=VPN root CA".  I was posting the question thinking this was a residual setting I had from when I first set up StrongSwan in the AM and successfully connected from my windows box.  I then pivoted and went to the S2S config and those were the errors.

I was just trying to rule out that as the issue so that I could continue tweaking settings.  I will relook the configs so that I don't lock myself out and will review the example configs you posted.  Unfortunately, StrongSwan is what that other company is using so we will have to continue until we get the settings correct.

Thanks for the link and suggestion,

**Comment by antai1** [comment]
I was able to tweak the config and get the tunnel to come up just having to
deal with routing issue.  Can't ping from one side to the other.

Thanks for the help.


V/r


Israel P. Anta

Nova Technologies

SOMPE-G Principal Support Engineer

ARSOF Mission Planning Support - Depot

Bldg 3-1424 Modularity Road

Fort Bragg, NC 28310

Office: 910-907-0642 DSN: 337-0642

Desk: 910-432-5518 Cell: 910-682-0692

***@***.***

***@***.***

***@***.***>

**Reply by Thermi** [reply]
Okay. Is the issue resolved? If so, please select your answer as having answered the question.

**Comment by antai1** [comment]
Yes, the issue was resolved.  Thanks for the assistance.

**Comment by antai1** [comment]
Yes, it is.

V/r


Israel P. Anta

Nova Technologies

SOMPE-G Principal Support Engineer

ARSOF Mission Planning Support - Depot

Bldg 3-1424 Modularity Road

Fort Bragg, NC 28310

Office: 910-907-0642 DSN: 337-0642

Desk: 910-432-5518 Cell: 910-682-0692

***@***.***

***@***.***

***@***.***>
